lib/spack/spack/test/env.py

```diff
@@ -385,3 +378,40 @@ def test_can_add_specs_to_environment_without_specs_attribute(tmp_path, mock_pac
 
     assert len(env.user_specs) == 1
     assert env.manifest.pristine_yaml_content["spack"]["specs"] == ["a"]
+
+
+@pytest.mark.parametrize("filename", [ev.lockfile_name, "as9582g54.lock", "m3ia54s.json"])
+def test_initialize_from_lockfile(tmp_path, filename):
```

Comment:
I would add:
```suggestion
@pytest.mark.regression("37410")
def test_initialize_from_lockfile(tmp_path, filename):
```
to each test, so we have a quick reference of the regression

Reply:
`test_cannot_initialize_from_bad_lockfile` is not strictly a result of the regression. It tests code added in this PR which improves robustness but is not directly related to #37410.